test/jobs/calendar_importer/resident_advisor_importer_test.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class ResidentAdvisorParserTest < ActiveSupport::TestCase
+  test 'detect if its a club' do
+    parser = CalendarImporter::Parsers::ResidentAdvisor
+    assert_equal [:clubs,  182_584], parser.ra_entity('https://ra.co/clubs/182584')
+  end
+
+  test 'detect if its a promoter' do
+    parser = CalendarImporter::Parsers::ResidentAdvisor
+    assert_equal [:promoters,  133_684], parser.ra_entity('https://ra.co/promoters/133684')
+  end
+
+  test 'reject if its not a promoter or club' do
+    parser = CalendarImporter::Parsers::ResidentAdvisor
+    assert_not parser.ra_entity('https://ra.co/promarters/133684')
+    assert_not parser.ra_entity('https://ras.co/clubs/133684')
+    assert_not parser.ra_entity('https://ra.co/133684')
+    assert_not parser.ra_entity('https://ra.co/asdasd/')
+  end
+  # test 'extracts events from Eventbrite calendars' do
```

Comment:
Do you want to keep this?

Reply:
this is a WIP PR, ill request review when im ready :)